pkg/config/config_template.yaml

```diff
@@ -1,424 +1,781 @@
-# If running dogstatsd in a container, host PID mode (e.g. with --pid=host) is required.
-# dogstatsd_origin_detection: false
-#
-# The buffer size use to receive statsd packet, in bytes
-# dogstatsd_buffer_size: 8192
-#
-# Whether dogstatsd should listen to non local UDP traffic
-# dogstatsd_non_local_traffic: false
-#
-# Publish dogstatsd's internal stats as Go expvars
-# dogstatsd_stats_enable: false
-#
-# How many items in the dogstatsd's stats circular buffer
-# dogstatsd_stats_buffer: 10
-#
-# The port for the go_expvar server
-# dogstatsd_stats_port: 5000
-#
-# The number of bytes allocated to dogstatsd's socket receive buffer (POSIX
-# system only). By default, this value is set by the system. If you need to
-# increase the size of this buffer but keep the OS default value the same, you
-# can set dogstatsd's receive buffer size here. The maximum accepted value
-# might change depending on the OS.
-# dogstatsd_so_rcvbuf:
-#
-# When this flag is enabled, Dogstatsd collects basic statistics (count/last seen)
-# about the metrics it processsed. Use the agent command "dogstatsd-stats" to visualize
-# those statistics. Note that this feature slightly increase the cpu usage
-# of the dogstatsd server. Set this flag to true to activate it.
-# dogstatsd_metrics_stats_enable: false
-#
-# Additional tags to append to all metrics, events and service checks received by
-# this dogstatsd server. Useful for tagging all dogstatsd metrics reporting from
-# a single host without resorting to host tags.
-# dogstatsd_tags:
-#   - name:value
-#
-# If you want to forward every packet received by the dogstatsd server
-# to another statsd server, uncomment these lines.
-# WARNING: Make sure that forwarded packets are regular statsd packets and not "dogstatsd" packets,
-# as your other statsd server might not be able to handle them.
-# statsd_forward_host: address_of_own_statsd_server
-# statsd_forward_port: 8125
-#
-# If you want all statsd metrics coming from this host to be namespaced
-# you can configure the namspace below. Each metric received will be prefixed
-# with the namespace before it's sent to Datadog.
-# statsd_metric_namespace:
+
+##################################
+## DogSatsD Agent Configuration ##
+##################################
+
+## @param use_dogstatsd - boolean - optional - default: true
+## Set this option to false to disable the Agent DogStatsD server.
+#
+#  use_dogstatsd: true
+
+## @param dogstatsd_port - integer - optional - default: 8125
+## Override the Agent DogStatsD port.
+## Note: Make sure your client is sending to the same UDP port.
+#
+#  dogstatsd_port: 8125
+
+## @param bind_host - string - optional - default: localhost
+## The host to bind to receive external metrics (used only by the DogStatsD server for now).
+## For DogStatsD this is ignored if 'dogstatsd_non_local_traffic' is set to true.
+#
+#  bind_host: localhost
+
+## @param dogstatsd_socket - string - optional - default: /var/run/dogstatsd/dsd.sock
```

Comment:
This is optional https://github.com/DataDog/datadog-agent/blob/440eb3344cb073eae546be2d1535eb73ed617e9b/pkg/config/config.go#L212